libraries/botbuilder-dialogs/src/waterfallDialog.ts

```diff
@@ -88,6 +88,12 @@ export class WaterfallDialog<O extends object = {}> extends Dialog<O> {
         }
     }
 
+    public get version(): string {
+        // Simply return the id + number of steps to help detect when new steps have
+        // been added to a given waterfall.
+        return `${this.id}:${this.steps.length}`;
+    }
```

Comment:
I wonder if this should be a method.  

This requires that the langauge supports virtual properties.  For example, WaterfallDialog derives from Dialog and so needs to "override" the base version because it's a property.